### PR TITLE
N911 9555

### PR DESCRIPTION
**Pull request by Ciejo**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -71,6 +71,17 @@ resource "aws_instance" "this" {
   }
 }
 
+resource "aws_ebs_volume" "volume-swap" {
```
Comment (ivours) [inline main.tf]: Resource type not needed in the name.

Instead of `volume-swap`, we can name it just  `swap`

File: main.tf [hunk 2]
```diff
@@ -71,6 +71,17 @@ resource "aws_instance" "this" {
...
+  availability_zone = data.aws_availability_zones.available.names[0]
+  size              = var.size
+}
+
+resource "aws_volume_attachment" "ebs_att" {
```
Comment (ivours) [inline main.tf]: We can name it `swap`

File: main.tf (hunk 1 above)
Comment (ivours) [reply main.tf]: Also update references

File: variables.tf [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
+variable "private_subnet_ids" {}
 variable "vpc_id" {}
 variable "ami_id" {}
 variable "instance_type" {}
 variable "ingress_cidr_blocks" {}
 variable "zone_id" {}
 variable "email" {}
 variable "dns" {}
-
+variable "size" {}
```
Comment (ivours) [inline variables.tf]: A more descriptive name for this variable should be `volume_size` (also update references)

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+POLICY
+}
+
+resource "aws_iam_role_policy" "this" {
+  name   = "${var.environment}-gitlab"
+  role   = aws_iam_role.this.id
+  policy = <<-EOF
+{
+    "Version": "2012-10-17",
+    "Id": "certbot-dns-route53 sample policy",
```
Comment (ivours) [inline main.tf]: You can remove this id

File: resources/templates/docker-compose.yml.tpl [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    image: 'gitlab/gitlab-ce:latest'
+    hostname: '${hostname}'
+    environment:
+      GITLAB_OMNIBUS_CONFIG: |
+        external_url 'https://${hostname}'
+        letsencrypt['enable'] = false
+        nginx['ssl_certificate'] = "/etc/gitlab/ssl/${hostname}.crt"
+        nginx['ssl_certificate_key'] = "/etc/gitlab/ssl/${hostname}.key"
+    ports:
+      - '2222:22'
```
Comment (ivours) [inline resources/templates/docker-compose.yml.tpl]: We should use 22:22 for Gitlab SSH.

Also, please change default SSH port to the EC2 instance, to port 23, for example.

File: main.tf (hunk 1 above)
Comment (Ciejo) [reply main.tf]: Done, the new name is volume-swap.

File: main.tf (hunk 2 above)
Comment (Ciejo) [reply main.tf]: Done, changed to swap.

File: variables.tf (hunk 3 above)
Comment (Ciejo) [reply variables.tf]: Done, changed to volume_size

File: main.tf (hunk 4 above)
Comment (Ciejo) [reply main.tf]: Done, the id was removed.

File: resources/templates/docker-compose.yml.tpl (hunk 5 above)
Comment (Ciejo) [reply resources/templates/docker-compose.yml.tpl]: Done. The gitlab ssh uses 22:22 and the port to the EC2 instance is 2222

File: variables.tf [hunk 6]
```diff
@@ -8,4 +8,4 @@ variable "ingress_cidr_blocks" {}
 variable "zone_id" {}
 variable "email" {}
 variable "dns" {}
-variable "size" {}
+variable "volume_size" {}
```
Comment (ivours) [inline variables.tf]: I'd name it `swap_volume_size`

File: main.tf [hunk 7]
```diff
@@ -37,11 +37,23 @@ module "security_group_gitlab" {
...
   vpc_id              = var.vpc_id
-  ingress_cidr_blocks = var.ingress_cidr_blocks
+  ingress_cidr_blocks = [var.ingress_cidr_blocks]
   ingress_rules       = ["https-443-tcp", "ssh-tcp", "openvpn-udp"]
+  ingress_with_cidr_blocks = [
+    {
+      from_port   = 2222
+      to_port     = 2222
+      protocol    = "tcp"
+      description = "New SSH"
```
Comment (ivours) [inline main.tf]: Probably "SSH port override for host OS" would me more descriptive

File: variables.tf (hunk 6 above)
Comment (Ciejo) [reply variables.tf]: Done. Changed to swap_volume_size

File: main.tf (hunk 7 above)
Comment (Ciejo) [reply main.tf]: Done. Changed to "SSH port override for host OS" would me more descriptive

**Comment by ivours** [comment]
@simple-me I see the `santi` keyword in some of the resources name, please remove it.

Thanks!

File: main.tf (hunk 1 above)
Comment (ivours) [reply main.tf]: Please name it just `swap`

Comment (Ciejo) [reply main.tf]: Done, it was swap already. I put bad the name here.

**Pull request merged**